server/lib/tags.ts

```diff
@@ -0,0 +1,42 @@
+import { uniq } from 'lodash';
+
+export const validateTags = (tags: string[]): void => {
+  if (tags) {
+    // Limit to max 30 tags
+    if (tags.length > 30) {
+      throw new Error(`Sorry, you can't add more than 30 tags. Please remove ${30 - tags.length} tag(s).`);
+    }
+
+    // Validate each individual tags
+    tags.forEach(tag => {
+      if (tag.length === 0) {
+        throw new Error("Can't add empty tags");
+      } else if (tag.length > 32) {
+        throw new Error(`Tag ${tag} is too long, must me shorter than 32 characters`);
+      }
+    });
+  }
+};
+
+export function setTags(tags?: string[]) {
+  if (tags) {
+    tags = uniq(
+      tags
+        .map(tag => {
+          if (tag) {
+            const upperCase = tag.toUpperCase();
+            return upperCase.trim().replace(/\s+/g, ' ');
+          }
+        })
+        .filter(tag => {
+          return tag && tag.length > 0;
+        }),
+    );
+  }
+
+  if (!tags || tags.length === 0) {
+    this.setDataValue('tags', null);
+  } else if (tags) {
+    this.setDataValue('tags', tags);
+  }
```

Comment:
This doesn't look right to me, `setTags` has its own context so `this` here probably not refers to the models.

I'd prefer if we can rename the function `cleanTags`, return the value and do the `setDataValue` from the model

Reply:
The reason why `this` works in this context is because `setTags` is an actual `function`. If it was an arrow expression, `this` would be undefined.

Do you think this is too technical and we should just extract the functional expressions?

Reply:
> The reason why this works in this context is because setTags is an actual function. If it was an arrow expression, this would be undefined.

Indeed, that explains why it works :+1: 

> Do you think this is too technical and we should just extract the functional expressions?

I do, it's seems more useful to me to have a generic helper to clean tags that doesn't know anything about models and can be called from any code. It's also more practical to write unit tests if the function has no side effect.